### PR TITLE
#1601: Go to Your Feed tip

### PR DESCRIPTION
**Pull request by joshuatbrown**
## Issues covered
#1601 

## Description
Shows a "Go to your feed..." tip at the bottom of the Discover tab. It doesn't match the design, but getting the popover to work was impossible in the time I had with the knowledge I had. Maybe we can still make it happen.

~- [ ] try to use a real popover instead of this fake one~ Linda is fine with this approach
- [x] show the tip after the user has followed 3 accounts
- [x] don't show the tip to people who have been using the app forever
- [x] dismiss the tip when the user taps the Feed tab

## How to test
1. Navigate to Discover
2. Observe the tip at the bottom
3. If you dismiss it and want to test again, you can delete the app and reinstall. Or, for developer testing, add this line to `NosApp.init()` after the `Tips.configure()` line:

```
Tips.showAllTipsForTesting()
```

## Screenshots/Video

| Before | After |
|--------|--------|
| Simulator Screenshot - iPhone 16 Pro - 2024-10-11 at 17 22 39 | Simulator Screenshot - iPhone 16 Pro - 2024-10-11 at 17 22 36 |

### CODE REVIEW

**Comment by joshuatbrown** [comment]
> This seems tough but the emoji is a nice hack for now. Did you check with Linda somewhere to see if she is ok with iPhone-only?

I checked again, and it looks OK on iPad when building with Xcode 15.4. I still [asked](https://github.com/planetary-social/nos/issues/1601#issuecomment-2417694836) Linda to be sure.

> 
> Some thoughts:
> 
> * Would it be easy to fire some analytics events when each tip is displayed/dismissed? This would help us measure and optimize our onboarding flow in the future.

Not really. The only way I can find to send analytics events is when the TipView appears or disappears. But those happen any time you navigate to/from the tab where they're displayed, so you could get 100 "welcome tip displayed" and 100 "welcome tip disappeared" events in a session. Maybe that's fine? It tells us that the user hasn't dismissed it yet

> * Can you layer the discover tab scroll view _under_ this tip? That would match the Figma design more closely and look nicer I think.

I can try. But I'm concerned that it'll cover the bottom of the scroll view. Which should be fine. The user can easily dismiss it. Yeah, I'll work on this.

> * I expected that tapping on the home tab as the tip instructed would automatically dismiss the tip. I don't see a user story for this but does that make sense to you?

Yeah, it _should_ dismiss the tip as soon as the Feed tab is displayed. It used to. But maybe something broke (or was [fixed](https://github.com/planetary-social/nos/pull/1645)) and now you have to tap twice. I'll look into it.

> * Might be worth calling out to Linda that these tips will only run the first time the app is installed - not after logout. Some apps have a "Reset tips" button in the settings. That might be nice to add for our users and for testing.

Asking Linda in Slack.

File: Nos/Views/Home/HomeFeedView.swift
Comment (joshuatbrown) [inline Nos/Views/Home/HomeFeedView.swift:82]: @mplorentz bringing the discussion about analytics into this thread. Above, you can see what I did, but as I'm sure you'll see, this means we may send the "welcome feed displayed" event many many many times, as well as the dismissed one. Maybe that's worth something, and maybe it's better than nothing.

**Comment by joshuatbrown** [comment]
@mplorentz fixed the bug where the "Go to Feed" tip wasn't dismissed after going to the Feed and layered the tip over the scroll view. The last thing for you is the analytics question (above), and then there are two questions for Linda. After addressing the analytics, I propose we merge this so it's in a build that Linda can see and test for herself.

**Comment by joshuatbrown** [comment]
Linda says she doesn't need a Reset Tips feature for testing since she can often deletes and reinstalls the app for testing anyway. I vote we leave it out for now, see if users want/need/ask for it, then add it at that point.

**Comment by joshuatbrown** [comment]
Another update: [Linda is good](https://github.com/planetary-social/nos/issues/1601#issuecomment-2417807827) with the way this looks on iPad.

File: Nos/Views/Home/HomeFeedView.swift
Comment (mplorentz) [reply Nos/Views/Home/HomeFeedView.swift:82]: ok, let's not do this. I was thinking it would be nice to know what users do after seeing the tips. Do they follow them, do they stay in the app longer, etc. But we can't get that sort of data from this.

**Comment by mplorentz** [comment]
👀

**Review by mplorentz — APPROVED** [review]
LGTM! I approve pending removal of the analytics.

**Pull request merged**
